src/lib/expression/lqp_column_expression.cpp

```diff
@@ -43,7 +43,10 @@ std::string LQPColumnExpression::as_column_name() const {
 
 DataType LQPColumnExpression::data_type() const {
   const auto original_node = column_reference.original_node();
-  if (column_reference.original_node()->type == LQPNodeType::StoredTable) {
+  if (column_reference.original_column_id() == INVALID_COLUMN_ID) {
+    // Handle COUNT(*)
+    return DataType::Int;
```

Comment:
I found this comment in aggregate_sort.cpp:
```
As the input type does not matter and we do not even have an input column,
but the function call expects an input type, we choose Int arbitrarily.
This is NOT the result type of COUNT(*), which is Long.
```

Reply:
I see. This is the (fake) input type, not the output type. Could you add a comment?

Reply:
Wouldn't it be easier to replace Int with Long to avoid confusion? I think about a person who does not find these comments in the beginning. This way, we do not need the comments at all.

Reply:
Yes, feel free to replace it. I'd still add a comment here explaining that we are looking at the input data type.